mailpile/plugins/tags.py

```diff
@@ -361,29 +369,45 @@ def _do_tagging(self, ops, msg_ids, conversations, save=True, auto=False):
                 if conversation:
                     rv['conversations'] = True
 
+                ignored = False
                 tag_id = tag._key
+                tag_cfg = tag
                 tag = tag.copy()
                 tag["tid"] = tag_id
                 if op[0] == '-':
-                    removed = idx.remove_tag(self.session, tag_id,
-                                             msg_idxs=msg_ids,
-                                             conversation=conversation)
-                    rv['untagged'].append((tag, sorted([b36(i)
-                                                        for i in removed])))
+                    if force or tag_cfg['flag_allow_del']:
+                        removed = idx.remove_tag(self.session, tag_id,
+                                                 msg_idxs=msg_ids,
+                                                 conversation=conversation)
+                        rv['untagged'].append(
+                            (tag, sorted([b36(i) for i in removed])))
+                    else:
+                        rv['ignored'].append((op, tag))
+                        ignored = True
                 else:
-                    added = idx.add_tag(self.session, tag_id,
-                                        msg_idxs=msg_ids,
-                                        conversation=conversation)
-                    rv['tagged'].append((tag, sorted([b36(i)
-                                                      for i in added])))
+                    if force or tag_cfg['flag_allow_add']:
+                        added = idx.add_tag(self.session, tag_id,
+                                            msg_idxs=msg_ids,
+                                            conversation=conversation)
+                        rv['tagged'].append(
+                            (tag, sorted([b36(i) for i in added])))
+                    else:
+                        rv['ignored'].append((op, tag))
+                        ignored = True
+
                 # Record behavior
-                if len(msg_ids) < 15:
+                if len(msg_ids) < 15 and not ignored:
                     for t in self.session.config.get_tags(type='tagged',
                                                           default=[]):
                         idx.add_tag(self.session, t._key, msg_idxs=msg_ids)
             else:
                 self.session.ui.warning('Unknown tag: %s' % op)
 
+
+        if rv['ignored'] and (len(rv['tagged']) == len(rv['untagged']) == 0):
+            self.event.private_data['ignored'] = rv['ignored']
+            return self._error(_('Nothing Happened'), result=rv)
+
```

Comment:
leave it for later then.  Was just something I thought of reading the diff.  the way some other things do it is changing the cursor when the drop target is illegal, but that's for "all or nothing" cases again